addons/FlashCards/src/presenter.js

```diff
@@ -399,22 +400,21 @@ function AddonFlashCards_create(){
         presenter.showCard(presenter.state.currentCard);
     };
 
-    presenter.originalCard = null;
     presenter.nextCard = function (disregardNoLoop) {
-        if ((presenter.isFrontPlaying || presenter.isBackPlaying) && presenter.state.currentCard != presenter.originalCard) {
-            presenter.sendEndedEvent();
-        }
-        presenter.originalCard = presenter.state.currentCard;
         presenter.removeActiveElementClass();
         if (presenter.state.currentCard < presenter.state.totalCards){
             presenter.state.currentCard += 1;
         }else if (presenter.state.noLoop == false || disregardNoLoop){
             presenter.state.currentCard = 1;
         }
+        if ((presenter.isFrontPlaying || presenter.isBackPlaying) && presenter.state.currentCard != presenter.originalCard) {

```

Comment:
Miałem wątpliwości i przetestowałem na wersji testowej. 

Nie można w tym miejscu wykonywać sendEndedEvent bez przekazania parametru. sendEndedEvent korzysta z wartości presenter.state.currentCard, gdy nie poda się do metody zmiennej. W obecnej sytuacji, gdy gra audio z karty numer 1 i przejdzie się na kartę numer 2 to event "ended" będzie wysyłany wraz z "item": "2", mimo że nagranie z karty numer 1 się kończy, gdyż zaktualizowałeś presenter.state.currentCard zanim wysłałeś event.

Myślę, że albo warunek trzeba dopracować i powinien być sprawdzany na początku metody lub do sendEndedEvent przekazywać presenter.originalCard.